hs_core/models.py

```diff
@@ -1898,6 +1900,35 @@ def delete_all_elements(self):
         self.relations.all().delete()
         self.funding_agencies.all().delete()
 
+    def db_instance_to_dict(self, instance, fields=None, exclude=None):
+        """
+        modified version of Django's model_to_dict method accomodates many_to_many relations
+
+        Returns a dict containing the data in ``instance`` suitable for passing as
+        a Form's ``initial`` keyword argument.
+
+        ``fields`` is an optional list of field names. If provided, only the named
+        fields will be included in the returned dict.
+
+        ``exclude`` is an optional list of field names. If provided, the named
+        fields will be excluded from the returned dict, even if they are listed in
+        the ``fields`` argument.
+        """
+        opts = instance._meta
+        data = {}
+        for f in chain(opts.concrete_fields, opts.many_to_many):
```

Comment:
In case of SWAT Model Instance resource type - I think if we change the following method to check whether the parameter ```objectives``` is a list of strings or int (pk) and then use this parameter accordingly the issue should resolve.:

```
@classmethod
   def _add_swat_objective(cls, model_objective, objectives):